logrus.go

```diff
@@ -6,69 +6,85 @@ import (
 	"os"
 	"time"
 
-	"go.mozilla.org/mozlog"
-
 	"github.com/sirupsen/logrus"
 )
 
-// Enable enables mozlogrus
-func Enable(loggerName string) {
-	logrus.SetFormatter(&MozLogFormatter{
-		LoggerName: loggerName,
-	})
+var hostname string
+var pid int
 
-	logrus.SetOutput(os.Stdout)
-}
+func init() {
+	var err error
+	hostname, err = os.Hostname()
+	if err != nil {
+		hostname = "unknown"
+	}
 
-// SetFormatterName the LoggerName for the logrus formatter
-func SetFormatterName(name string) {
-	logrus.SetFormatter(&MozLogFormatter{
-		LoggerName: name,
-	})
+	pid = os.Getpid()
 }
 
-// prefixFieldClashes is from logrus package
-func prefixFieldClashes(data logrus.Fields) {
-	_, ok := data["msg"]
-	if ok {
-		data["fields.msg"] = data["msg"]
-	}
+// Enable changes the default logrus formatter to MozLogFormatter and
+// sets output to stdout
+func Enable(m *MozLogFormatter) {
```

Comment:
Let's leave `Enable` alone and add a `EnableWithFormatter(m *MozLogFormatter)`. This will maintain backwards compatibility. Enable was also meant to be a simple way of initializing logging with common options. `Enable` should also set Type to `app.log` since that is the current default.

Reply:
I think I've already broken backwards compatibility by removing some exports from the package, e.g: `AppLog => appLog`.  If somebody ports to using V2, it's tiny code change from V1 to V2.

Reply:
AppLog was not available before. I just copied it over in #6. I still like the idea of a simple constructor with good defaults.